jobsdb/jobsdb.go

```diff
@@ -4590,3 +4642,8 @@ func QueryWorkspacePileupWithRetries(parentContext context.Context, timeout time
 	}
 	return res.(map[string]map[string]int), err
 }
+
+type smallDS struct {
+	ds          dataSetT
+	recordsLeft int
+}
```

Comment:
**minor:** I would prefer declaring the `smallDS` type inside `migrateDSLoop`, as it really is private to that method